### PR TITLE
Correcting race condition in creating HPX ID maps in hpx dataflow simulator

### PR DESCRIPTION
**Pull request by zbyerly**

### CODE REVIEW

File: src/parallelization/hpxdataflowsimulator.h [hunk 1]
```diff
@@ -109,19 +112,22 @@ class CellComponent
         std::vector<hpx::future<void> > remoteIDFutures;
         remoteIDFutures.reserve(neighbors.size());
 
+	hpx::lcos::local::spinlock mutex;
```
Comment (gentryx) [inline src/parallelization/hpxdataflowsimulator.h]: hmm, above you pull in the standard mutex header, but here you instantiate a spinlock from HPX. Is the include wrong or this type or am I missing something?

**Comment by gentryx** [comment]
Some indentation looks odd, but I'll correct that post merge. This PR is good to go once I understood the which mutex type to use. Thanks so much for debugging and fixing this!

File: src/parallelization/hpxdataflowsimulator.h (hunk 1 above)
Comment (hkaiser) [inline src/parallelization/hpxdataflowsimulator.h]: The #include is for the `std::lock_guard<>`.

Comment (gentryx) [inline src/parallelization/hpxdataflowsimulator.h]: @hkaiser d'oh, thanks!

**Pull request merged**